app/server/api/credit_transfer_api.py

```diff
@@ -145,11 +145,21 @@ def put(self, credit_transfer_id):
             }
             return make_response(jsonify(response_object)), 400
 
-        if action == 'COMPLETE':
-            credit_transfer.resolve_as_complete_and_trigger_blockchain()
+        try:
+            if action == 'COMPLETE':
+                credit_transfer.resolve_as_complete_and_trigger_blockchain()
+
+            elif action == 'REJECT':
+                credit_transfer.resolve_as_rejected()
+
+        except Exception as e:
+
+            db.session.commit()
```

Comment:
why db.session.commit() on a 400? To submit the error to the db?

Reply:
if a txn fails completion due to a limit, it's rejected and saved